internal/cosesign1/cosesign1util_test.go

```diff
@@ -0,0 +1,153 @@
+//go:build linux
+// +build linux
+
+package cosesign1
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/veraison/go-cose"
+)
+
+func readFileBytes(filename string) []byte {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		println("Error reading '" + filename + "': " + string(err.Error()))
+	}
+	if len(content) == 0 {
+		println("Warning: empty file '" + filename + "'")
+	}
+	return content
+}
+
+func readFileString(filename string) string {
+	return string(readFileBytes(filename))
+}
+
+var FragmentRego string
+var FragmentCose []byte
+var LeafPrivatePem string
+var LeafCertPEM string
+var LeafPubkeyPEM string
+var CertChainPEM string
+
+func comparePEMs(pk1pem string, pk2pem string) bool {
+	pk1der := pem2der([]byte(pk1pem))
+	pk2der := pem2der([]byte(pk2pem))
+	return bytes.Equal(pk1der, pk2der)
+}
+
+/*
+	Decode a COSE_Sign1 document and check that we get the expected payload, issuer, keys, certs etc.
+*/
+
+func Test_UnpackAndValidateCannedFragment(t *testing.T) {
+	var unpacked UnpackedCoseSign1
+	unpacked, err := UnpackAndValidateCOSE1CertChain(FragmentCose, nil, nil, false)
+
+	if err != nil {
+		t.Errorf("UnpackAndValidateCOSE1CertChain failed: %s", err.Error())
+	}
+
+	var iss = unpacked.Issuer
+	var feed = unpacked.Feed
+	var pubkey = base64PublicKeyToPEM(unpacked.Pubkey)
+	var pubcert = base64CertToPEM(unpacked.Pubcert)
+	var payload = string(unpacked.Payload[:])
+	var cty = unpacked.ContentType
```

Comment:
more idomatic (you can apply this elsewhere as well)
```suggestion
	iss := unpacked.Issuer
	feed := unpacked.Feed
	pubkey := base64PublicKeyToPEM(unpacked.Pubkey)
	pubcert := base64CertToPEM(unpacked.Pubcert)
	payload := string(unpacked.Payload[:])
	cty := unpacked.ContentType
```